core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1090,13 +1092,41 @@ class KafkaApis(val requestChannel: RequestChannel,
         )
       }
 
+      def handleError(e: Throwable): Unit = {
+        error(s"Unexpected error handling request ${request.requestDesc(true)} " +
+          s"with context ${request.context}", e)
+        requestHelper.handleError(request, e)
+      }
+
       if (ElasticLogManager.enabled()) {
         // The fetching is done is a separate thread pool to avoid blocking io thread.
-        fetchingExecutors.submit(new Runnable {
+        fastFetchExecutor.submit(new Runnable {
           override def run(): Unit = {
-            ReadAllHint.mark()
-            doFetchingRecords()
-            ReadAllHint.reset()
+            try {
+              ReadHint.markReadAll()
+              ReadHint.markFastRead();
+              doFetchingRecords()
+              ReadHint.clear()
+            } catch {
+              case e: Throwable =>
+                val ex = FutureUtil.cause(e)
+                val fastReadFailFast = ex.isInstanceOf[FastReadFailFastException]
+                if (fastReadFailFast) {
+                  slowFetchExecutor.submit(new Runnable {
+                    override def run(): Unit = {
+                      try {
+                        ReadHint.markReadAll()
+                        doFetchingRecords()
```

Comment:
ReadHint.clear()

Reply:
The thread is only used for fetch, it's ok not clear